src/hosted_agents/hosted_agents_starter.js

```diff
@@ -26,3 +24,13 @@ function register_rpc() {
         logging: true,
     });
 }
+
+async function start_hosted_agents() {
+    mongo_client.instance().connect();
+    register_rpc();
+
+    // Try to start the hosted agents metrics server
+    await prom_reporting.start_server(config.HA_METRICS_SERVER_PORT);
+}
+
+start_hosted_agents();
```

Comment:
In Endpoint, you want on it. Here not. 
Is there a reason?

Reply:
I assume you mean "wait on it" T
We never wait in the global scope because NodeJS does not allow it (in the current version)
But even if it will allow it I see no reason. Starting a service is "call and forget about it" thingy